app/views/components/gds/timeline.scala.html

```diff
@@ -0,0 +1,26 @@
+
+@import java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME
+
+@import views.helpers.TimelineEvent
+
+@this()
+
+@(timelineEvents: Seq[TimelineEvent])
+
+@htmlForTimelineEvent(timelineEvent: TimelineEvent) = {
+    <li class="hmrc-timeline__event">
+        <h2 class="hmrc-timeline__event-title">@timelineEvent.title</h2>
+        <time class="hmrc-timeline__event-meta" datetime="@{ISO_OFFSET_DATE_TIME.format(timelineEvent.dateTime)}">
+            @timelineEvent.dateTimeAsShown
+        </time>
+        @timelineEvent.content.map { content =>
+            <div class="hmrc-timeline__event-content">
+                @content
+            </div>
+        }
+    </li>
+}
+
+<ol class="hmrc-timeline govuk-!-margin-top-5">
+    @timelineEvents.map(htmlForTimelineEvent)
+</ol>
```

Comment:
Have you checked if there is a component available in govuk/hmrc frontend libs?

Reply:
The timeline only consists of CSS styling, it's not defined as a component.
If you think about it, different services might have different requirements in term of what to display in the timeline (even if the design guidelines ask for at least title, date and/or time) and how to style the timeline's items, so IMO there is not point to have it as a component from the library POV. There are only two other hmrc services in github using the timeline CSS and they are also defining their own timeline component.

Reply:
You are right. And according to what you're saying, there are not much adoption for this pattern.
But I think it is feasible to do such a generic component that could serve all services.

Anyway, if it is not available ATM, then we have no other choice.